### DISCUSSION TITLE
Ability to use variables in yaml sequentially

### DISCUSSION THREAD

**Discussion by servohatred** [artilleryio/artillery#2373] [Artillery]
I wonder if there is any chance to get something similar of what we have for payloads but in variables. They are intended to work the same but variables are not as configurable as payloads are. I am talking about the "order" field  https://www.artillery.io/docs/reference/test-script#example

I am trying to run a scenario but using different values for a variable, the issue comes with the randomness of the "variables" field , we cannot control if all the variables are going to be used in the run, that meaning that I  cannot trust if all the scenarios were run at least once. I have tried using before scenario to set up the variables via js under the scenario object but it seems to not be working with playwright (the before step is never executed) 

Thanks in advance

**Comment by hassy** [comment]
Hi @servohatred 👋 If you want to work through a predefined list of values you can use a variable that contains a list, and then work through that list with `loop` with `over`. There's an example here: https://www.artillery.io/docs/reference/engines/http#looping-through-an-array
